src/main/java/Duke.java

```diff
@@ -1,10 +1,124 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+/**
+ * Main Duke chatbot class.
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    /** List of Tasks */
+    private ArrayList<Task> list;
+    private Duke(ArrayList<Task> list){
+
+        this.list = list;
+
+    }
+
+    public void greeting(){
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    public void add(String input) throws DukeEmptyCommandException{
+        if(input.equals("todo")){
+            throw new DukeEmptyCommandException("OOPS!!! The description of a todo cannot be empty.");
+        } else if (input.equals("deadline")){
+            throw new DukeEmptyCommandException("OOPS!!! The description of a deadline cannot be empty.");
+        } else if (input.equals("event")){
+            throw new DukeEmptyCommandException("OOPS!!! The description of an event cannot be empty.");
+        } else {
+            Task t = new Task(input);
+            this.list.add(t);
+            System.out.println("Got it. I've added this task: ");
+            System.out.println(t.toString());
+
+        }
+    }
+
+    public void display(){
+        System.out.println("Here are the tasks in your list:");
+        int len = this.list.size();
+                for(int i = 0; i < len; i++){
+                    int index = i + 1;
+                    Task t = this.list.get(i);
+                    if(t instanceof Todo){
+                        Todo todo = (Todo) t;
+                        System.out.println(index + "." + todo.toString());
+                    } else if ( t instanceof Event){
+                        Event event = (Event) t;
+                        System.out.println(index + "." + event.toString());
+                    } else if ( t instanceof Deadline){
```

Comment:
```suggestion
                for (int i = 0; i < len; i++) {
                    int index = i + 1;
                    Task t = this.list.get(i);
                    if (t instanceof Todo) {
                        Todo todo = (Todo) t;
                        System.out.println(index + "." + todo.toString());
                    } else if ( t instanceof Event) {
                        Event event = (Event) t;
                        System.out.println(index + "." + event.toString());
                    } else if ( t instanceof Deadline) {
```
I think you need some white space here